### PR TITLE
Improve suggested packages handling + testthat code update

### PR DESCRIPTION
**Pull request by olivroy**
# Summary
This PR aims to take advantage of [cli inline markup](https://cli.r-lib.org/reference/inline-markup.html) and rlang utils functions (i.e. `rlang::check_installed()` as well as using `dplyr::tibble()` in the code and documentation. 
Improving syntax of some error messages and improve speed of tests by using `expect_named/length/match/in()`

The failing tests on my Windows machine are related to file creation /saving. 

# Checklist

- [x] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [x] I have listed any major changes in the [NEWS](https://github.com/rstudio/gt/blob/master/NEWS.md). (No major change)
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality. N/A

Edit: sorry for the large PR. At first, I wanted to keep it small, but ended up making changes in all places I thought were relevant for consistency and to avoid many follow-up PRs)

### CODE REVIEW

**Comment by rich-iannone** [comment]
Sorry for not commenting sooner but I do like all the work you've put into this!

I have just one change request at the moment, and addressing that would get this PR closer to merging/completion. Would it be possible to revert all of the `tibble::tibble` changes back to `dplyr::tibble`? I know this doesn't make much sense, but I'm hoping to remove the **tibble** dependency at a later date (I use only a few **tibble** functions) and I know that **dplyr** will always re-export the `tibble()` function.

**Comment by olivroy** [comment]
@rich-iannone Thank you!

I have added a few more package handling changes + removed tibble::tibble in tests. + modernized a little the test suite.

File: tests/testthat/test-data_color.R [hunk 1]
```diff
@@ -1575,8 +1571,7 @@ test_that("The correct color values are obtained when using a color fn", {
     selection_value("style") %>%
     gsub("(background-color: |; color: .*)", "", .) %>%
     unique() %>%
-    length() %>%
-    expect_equal(3)
+    expect_length(4)
```
Comment (olivroy) [inline tests/testthat/test-data_color.R:1574]:
```suggestion
    expect_length(3)
```

**Comment by rich-iannone** [comment]
Thank you for being so diligent with the test changes (and checking the difference in timings). This helps a lot! Let me know when you want me to review and merge; I don’t want to do it too early, as you might have a few more commits to go until you consider it finalized.

**Comment by olivroy** [comment]
Hi @rich-iannone, I am pretty much done. Maybe to avoid back and forth, I can review one last time tomorrow. I will ping you once this is ready for review! (most likely Monday)

**Comment by olivroy** [comment]
@rich-iannone Ready for your review now.

**Review by rich-iannone — APPROVED** [review]
Thanks again for the huge amount of work done here! I made only a few minor changes but 99.5% of this was perfect.

**Pull request merged**